README.md

```diff
@@ -1,2 +1,96 @@
-# node-firefox-forward-ports
-Forward local ports to remote debugging interfaces on connected Firefox OS devices
+# node-firefox-forward-ports [![Build Status](https://secure.travis-ci.org/mozilla/node-firefox-forward-ports.png?branch=master)](http://travis-ci.org/mozilla/node-firefox-forward-ports)
+
+> Forward local ports to remote debugging interfaces on connected Firefox OS devices
+
+[![Install with NPM](https://nodei.co/npm/node-firefox-forward-ports.png?downloads=true&stars=true)](https://nodei.co/npm/node-firefox-forward-ports/)
+
+This is part of the [node-firefox](https://github.com/mozilla/node-firefox) project.
+
+Firefox OS devices offer a [Remote Debugging service](https://wiki.mozilla.org/Remote_Debugging_Protocol) 
+on a local network socket. This can be forwarded to a host computer over USB
+using ADB. This module takes care of that as necessary, reusing forwarded ports
+when possible.
+
+## Installation
+
+### From git
+
+```bash
+git clone https://github.com/mozilla/node-firefox-forward-ports.git
+cd node-firefox-forward-ports
+npm install
+```
+
+If you want to update later on:
+
+```bash
+cd node-firefox-forward-ports
+git pull origin master
+npm install
+```
+
+### npm
+
+```bash
+npm install node-firefox-forward-ports
+```
+
+## Usage
+
+This example uses [node-firefox-find-devices](https://github.com/mozilla/node-firefox-find-devices)
+to first assemble a list of connected Firefox OS devices.
+
+```javascript
+var findDevices = require('node-firefox-find-devices');
+var forwardPorts = require('node-firefox-forward-ports');
+
+findDevices().then(function (devices) {
+  return forwardPorts({
+    devices: devices
+  });
+}).then(function(results) {
+  console.log(results);
+});
+
+[
+  {
+    "id": "3739ced5",
+    "type": "device",
+    "isFirefoxOS": true,
+    "ports": [
```

Comment:
For example, this is what I get after I manually forward two ports to the same socket:
```json
 {
  "id": "3739ced5",
  "type": "device",
  "isFirefoxOS": true,
  "ports": [
   {
    "serial": "3739ced5",
    "local": "tcp:8002",
    "remote": "localfilesystem:/data/local/debugger-socket",
    "port": "8002"
   },
   {
    "serial": "3739ced5",
    "local": "tcp:8003",
    "remote": "localfilesystem:/data/local/debugger-socket",
    "port": "8003"
   }
  ]
 }
```
Thinking I should at least delete the `serial` property, because it's redundant. The `port` will be important for Firefox client connections.

The `remote` property gets important if we want to expand this to, for example, forward [Marionette test ports](https://developer.mozilla.org/en-US/docs/Marionette_Test_Runner). Then, those will have `"remote": "tcp:2828"`, and we'll want to use that to filter & select appropriate port forwards.